### DISCUSSION TITLE
Host header?

### DISCUSSION THREAD

**Discussion by gachikuku** [mitmproxy/mitmproxy#7006] [General]
Im doing Portswigger Host header authentication bypass lab.
When using Burp I get a Host header in the Request that is made.
But I don't in mitmproxy. why?!

**Comment by mhils** [comment]
If there is a host header on the wire, you should see one in mitmproxy.

**Reply by Prinzhorn** [reply]
I assume this is about HTTP/2 (because the Burp UI always renders a HTTP/1.1 equivalent for editing), maybe we should render the pseudo headers?

HTTP/1.1

![image](https://github.com/mitmproxy/mitmproxy/assets/679144/d7366a08-6ca6-4a75-853c-2d1eeaf3954f)

HTTP/2.0

![image](https://github.com/mitmproxy/mitmproxy/assets/679144/dde9897a-3b94-459e-8ad7-2f2cb9b1d06e)

Chromium

![image](https://github.com/mitmproxy/mitmproxy/assets/679144/d3647b5d-851d-43cf-b7e4-73290b4742ba)

**Comment by gachikuku** [comment]
Thank you for the replies.

But I don't know what's wrong I don't see it still.


https://github.com/mitmproxy/mitmproxy/assets/11743804/bdfa4698-2170-4afe-8850-241d84135818


Note: Both requests are HTTP/2. 
I thought it had something to do with transparent mode. 
It seems that is not the case though.

**Reply by Prinzhorn** [reply]
HTTP/2 does not have a `Host` header, but uses an `:authority` pseudo header, which the mitmproxy UI does not render.

**Reply by Prinzhorn** [reply]
Burp makes this transparent to the user and renders a fake HTTP/1.1 version for you for easier editing.

**Reply by gachikuku** [reply]
I think it should render it, like you said.

**Reply by gachikuku** [reply]
In that case how do I edit it? 
I tried adding the Host header myself and I get a Protocol Error

**Reply by Prinzhorn** [reply]
https://portswigger.net/burp/documentation/desktop/http2/http2-normalization-in-the-message-editor

**Reply by Prinzhorn** [reply]
I've opened https://github.com/mitmproxy/mitmproxy/issues/7007
